### ISSUE TITLE
pytest gets really slow at around the 95% mark

### ISSUE THREAD

**Issue by SHxKM** [pytest-dev/pytest-xdist#600]
I wasn't sure whether this should just be a comment on #110, but eventually decided to open a new issue since in my case nothing is hanging, but things get seriously slow.

`pytest -vvv -n 4 -x` starts getting *real* slow only after it reaches around 95%, and only with a certain branch that I've been working on for a while. 

`main` branch:

```
95 passed in 35.21s
```

`other` branch:

```
118 passed in 281.41s (0:04:41)
```

Where 90% of the time spent in the `other` branch tests is during the last 4-5 tests (those same 4-5 tests also exist in the `main` branch).

This leads me to believe that something I introduced in a new test is causing this behavior. All my tests are passing in both branches so I'm not sure what's causing the huge difference. I will inspect during the weekend and update.

**Comment by tonybaloney** [comment]
I recommend running the changes through a profiler like `cProfile`. Or using the `pytest-profiling` extension to see where the holdup is

**Comment by SHxKM** [comment]
Actually the branch theory was wrong. I'm getting the same slowness at around the 95% mark even with the `main` branch. I'll reword my OP.

**Comment by tamaskakuszi** [comment]
Hey @SHxKM. Did you find any solution to solve this issue? We are struggling with the same error here.
